code/modules/clothing/clothing.dm

```diff
@@ -138,7 +141,34 @@
 		moth_snack.clothing = WEAKREF(src)
 	moth_snack.attack(attacker, user, params)
 
+/obj/item/clothing/proc/salvage(obj/item/W, mob/user, params)
+	if(!HAS_BLOOD_DNA(src))
+		switch(salvage_material)
+			if("cloth")
+				new /obj/item/stack/sheet/cotton/cloth(src.drop_location(), 3)
+			if("leather")
+				new /obj/item/stack/sheet/leather(src.drop_location(), 3)
+			if("durathread")
+				new /obj/item/stack/sheet/cotton/cloth/durathread(src.drop_location(), 3)
+	else
+		switch(salvage_material)
+			if("cloth")
+				new /obj/item/stack/sheet/cotton/cloth/bloody(src.drop_location(), 3)
+			if("leather")
+				new /obj/item/stack/sheet/leather(src.drop_location(), 3) //Bloody leather and durathread will exist soon
+			if("durathread")
+				new /obj/item/stack/sheet/cotton/cloth/durathread(src.drop_location(), 3)
+	playsound(loc, 'sound/items/wirecutter.ogg', 50, 1, -1)
+	qdel(src)
+
 /obj/item/clothing/attackby(obj/item/W, mob/user, params)
+	if((W.tool_behaviour == TOOL_WIRECUTTER || W.is_sharp()) && salvage_material != null && do_after(user, 1.5 SECONDS))
+		salvage(src)
+		user.visible_message("[user] cuts [src] into pieces of [salvage_material] with [W].", \
+		"<span class='notice'>You cut [src] into pieces of [salvage_material] with [W].</span>", \
+		"<span class='hear'>You hear cutting.</span>")
+		return TRUE
+
 	if(!istype(W, repairable_by))
 		return ..()
```

Comment:
This check should be done first since the is_sharp check may override this, making clothes impossible to repair if their repair material is sharp (idk if that happens anywhere, but it might at some point and is_sharp is a bigger set than repairable by)

Reply:
This is done!